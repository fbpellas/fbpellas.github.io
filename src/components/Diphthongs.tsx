import * as React from 'react';
import { IPA_DEFINITION } from '../constants';
import { diphthongs } from '../data/phonemes';
import { PhonemesTable } from './PhonemesTable';
import { Tooltip } from './Tooltip';

export interface DiphthongsProps { }

const Diphthongs: React.FC<DiphthongsProps> = (_props) => {
  return (
    <div className="block-2">
      <div className="article">
        <h3 className="h3-title">Diphthongs</h3>
        <div>
          Diphthongs are a combination of two vowel sounds. There are eight diphthongs in the{' '}
          <Tooltip definition={IPA_DEFINITION} word='IPA' />: aɪ, eɪ, ɔɪ, aʊ, ɪə, ʊə, əʊ, eə. However, only
          five sounds are produced in American English.
        </div>
        <PhonemesTable data={diphthongs} />
      </div>
    </div>
  );
};

export { Diphthongs }