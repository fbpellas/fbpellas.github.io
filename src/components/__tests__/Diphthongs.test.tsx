import React from 'react';
import { render } from '@testing-library/react';
import { Diphthongs } from '../Diphthongs';

test('Diphthongs', () => {
  const { container } = render(<Diphthongs />);
  expect(container).toMatchSnapshot()
});