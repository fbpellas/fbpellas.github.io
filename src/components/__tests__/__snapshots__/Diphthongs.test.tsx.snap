// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diphthongs 1`] = `
<div>
  <div
    class="block-2"
  >
    <div
      class="article"
    >
      <h3
        class="h3-title"
      >
        Diphthongs
      </h3>
      <div>
        Diphthongs are a combination of two vowel sounds. There are eight diphthongs in the
         
        <div
          class="text-inline bottom-dot"
        >
          IPA
        </div>
        : aɪ, eɪ, ɔɪ, aʊ, ɪə, ʊə, əʊ, eə. However, only five sounds are produced in American English.
      </div>
      <table>
        <thead>
          <tr>
            <th>
              <div
                class="text-center"
              >
                Phonemes
              </div>
            </th>
            <th
              class="text-center"
            >
              <div
                class="text-center"
              >
                Grapheme
              </div>
            </th>
            <th
              class="text-center"
            >
              Examples
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              /aɪ/
            </td>
            <td>
              i, igh, y, ie
            </td>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  <u>
                    i
                  </u>
                  ce
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  b
                  <u>
                    ye
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  l
                  <u>
                    i
                  </u>
                  me
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  f
                  <u>
                    igh
                  </u>
                  t
                </div>
                ,
                 
              </div>
              <div
                class="text-inline nowrap"
              >
                sk
                <u>
                  y
                </u>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              /eɪ/
            </td>
            <td>
              a, a-e, ai, ay, eigh, ey
            </td>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  <u>
                    a
                  </u>
                  corn
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  j
                  <u>
                    a
                  </u>
                  d
                  <u>
                    e
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  p
                  <u>
                    ai
                  </u>
                  d
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  w
                  <u>
                    eigh
                  </u>
                  t
                </div>
                ,
                 
              </div>
              <div
                class="text-inline nowrap"
              >
                h
                <u>
                  ey
                </u>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              /ɔɪ/
            </td>
            <td>
              oi, oy
            </td>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  b
                  <u>
                    oy
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  c
                  <u>
                    oi
                  </u>
                  n
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  s
                  <u>
                    oi
                  </u>
                  l
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  v
                  <u>
                    oi
                  </u>
                  ce
                </div>
                ,
                 
              </div>
              <div
                class="text-inline nowrap"
              >
                turqu
                <u>
                  oi
                </u>
                se
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              /oʊ/
            </td>
            <td>
              o, ow
            </td>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  g
                  <u>
                    o
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  t
                  <u>
                    o
                  </u>
                  tal
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  gr
                  <u>
                    ow
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  <u>
                    o
                  </u>
                  ver
                </div>
                ,
                 
              </div>
              <div
                class="text-inline nowrap"
              >
                g
                <u>
                  ol
                </u>
                d
              </div>
            </td>
          </tr>
          <tr>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              /aʊ/
            </td>
            <td>
              ow
            </td>
            <td>
              <svg
                class="play-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  c
                  <u>
                    ow
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  n
                  <u>
                    ow
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  br
                  <u>
                    ow
                  </u>
                  n
                </div>
                ,
                 
              </div>
              <div
                class="text-inline"
              >
                <div
                  class="text-inline nowrap"
                >
                  w
                  <u>
                    ow
                  </u>
                </div>
                ,
                 
              </div>
              <div
                class="text-inline nowrap"
              >
                <u>
                  gow
                </u>
                n
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
